### PR TITLE
Neck Control Teleop Panel

### PR DESCRIPTION
**Pull request by wxmerkt**
Sends JOINT_POSITION_GOAL message for [valkyrie_translator](https://github.com/openhumanoids/valkyrie_translator)

image

### CODE REVIEW

File: src/python/director/teleoppanel.py [hunk 1]
```diff
@@ -1133,6 +1133,25 @@ def buildTabWidget(self, jointGroups):
...
+                    msg.num_joints = len(joints)
+                    msg.joint_name = joints
+                    msg.joint_velocity = [0] * len(joints)
+                    msg.joint_effort = [0] * len(joints)
+                    msg.joint_position = [0] * len(joints)
+                    for i, jointName in enumerate(joints):
+                        jointIndex = self.toJointIndex(jointName)
+                        msg.joint_position[i] = self.getJointValue(jointIndex)
+
+                    lc.publish("JOINT_POSITION_GOAL", msg.encode())
```
Comment (patmarion) [inline src/python/director/teleoppanel.py]: can you use:

    lcmUtils.publish("JOINT_POSITION_GOAL", msg)

File: src/python/director/teleoppanel.py [hunk 2]
```diff
@@ -1133,6 +1133,25 @@ def buildTabWidget(self, jointGroups):
                 self.slidersMap[jointName] = slider
                 self.labelMap[slider] = numericLabel
 
+            if groupName == 'Neck':
+                def onSendNeckJointPositionGoal():
+                    lc = lcm.LCM()
```
Comment (patmarion) [inline src/python/director/teleoppanel.py]: remove

**Comment by patmarion** [comment]
looks good to me.  see my comment about lcmUtils.  Thanks!

**Comment by wxmerkt** [comment]
Thanks for the super-quick review and comments. I've addressed these and also removed incorrect lcm usage at another place in the same file.

**Comment by mauricefallon** [comment]
Can you use bot_core_joint_angles_t.lcm instead
also want to remove neck_pitch_t from the multisense panel and use the same message

In two operation modes:
- typical mode: send this message and lcm2rosControl feeds it through
- whole body planning incl neck: this message would not be sent from ui and a high level controller would

**Comment by patmarion** [comment]
oops, sorry, just merged.  you could add commits and open a new PR.

**Comment by mauricefallon** [comment]
also can you separate the neck control buttons from the camera/lidar controls on the multisense tab
with two separate buttons.

Use DESIRED_NECK_ANGLES at channel

finally, all these changes should be legit for (1) atlas: real, mit_sim, ihmc_sim and (2) val: real and ihmc_sim. It shouldnt be too hard though

**Pull request merged**